### PR TITLE
fix(repo): Ensure build runs before lint in CI

### PR DESCRIPTION
**Pull request by BRKalow**
## Description

Our lint step is type-aware, and our internal dependencies rely on packages being built for types to be correctly available. I've updated the lint step in CI to rely on `build` so that the types should be available as expected.

I've also shifted the turbo cache interaction into our shared setup action. We can opt-in to restoring the turbo cache by passing `use-turbo-cache: 'true'`.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
       "outputs": []
     },
     "lint": {
-      "dependsOn": [],
+      "dependsOn": ["build"],
```
Comment (BRKalow) [inline turbo.json:66]: Several lint rules depend on types being available for local packages, this ensures they are. Going forward, we might want to split linting into type-aware and not type-aware to optimize this (cc @nikosdouvlis).

Comment (nikosdouvlis) [reply turbo.json:66]: Spot on Bryce. Expanding a little on the above, the general idea is that we want cheap jobs to run and possibly fail early, before other more expensive operations like package building, unit and integration tests etc.

**Review by nikosdouvlis — APPROVED** [review]
Very good job spotting this @BRKalow

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**